crm_compassion/model/event_compassion.py

```diff
@@ -114,65 +129,82 @@ def _get_year(self, cr, uid, ids, field_name, arg, context=None):
-        if event.lead_id:
-            event.lead_id.write({'event_id': new_id})
 
-        project_id = self._create_project(cr, uid, event, context)
-        analytic_id = self.pool.get('project.project').browse(
-            cr, uid, project_id, context).analytic_account_id.id
-        origin_obj = self.pool.get('recurring.contract.origin')
-        # Avoid putting twice the date in origin name
-        event_year = event.start_date[:4]
-        if event.name[-4:] == event_year:
-            event_year = ""
-        else:
-            event_year = " " + event_year
-        origin_id = origin_obj.create(cr, uid, {
-            'name': event.name + event_year,
-            'type': 'event',
-            'event_id': new_id,
-            'analytic_id': analytic_id,
-        }, context)
-        event.write({
+        # Create project for the tasks
+        project_id = False
+        if event.use_tasks:
+            project_id = self.pool.get('project.project').create(
+                cr, uid, self._get_project_vals(
+                    cr, uid, event, context), context)
+
+        # Analytic account and Origin linked to this event
+        analytic_id = self.pool.get('account.analytic.account').create(
+            cr, uid, self._get_analytic_vals(
+                cr, uid, event, context),
+            context)
+        origin_id = self.pool.get('recurring.contract.origin').create(
+            cr, uid, self._get_origin_vals(
+                cr, uid, event, analytic_id, context), context)
+        super(event_compassion, self).write(cr, uid, event.id, {
             'origin_id': origin_id,
             'analytic_id': analytic_id,
             'project_id': project_id,
-        })
+        }, context)
         return new_id
 
     def write(self, cr, uid, ids, vals, context=None):
-        """ Push values to linked project. """
+        """ Push values to linked objects. """
         super(event_compassion, self).write(cr, uid, ids, vals, context)
-        project_vals = dict()
-        ctx = context.copy()
-        ctx['from_event'] = True
-        if 'type' in vals:
-            project_vals.update({'project_type': vals['type']})
-        if 'user_id' in vals:
-            project_vals.update({'user_id': vals['user_id']})
-        if 'partner_id' in vals:
-            project_vals.update({'partner_id': vals['partner_id']})
-        if 'name' in vals:
-            project_vals.update({'name': vals['name']})
-        if project_vals:
-            for event in self.browse(cr, uid, ids, context):
-                if 'type' in vals:
-                    # Change parent of analytic account
-                    project_vals.update({
-                        'parent_id': self._find_parent_analytic(
-                            cr, uid, vals['type'], event.year, context)})
-                event.project_id.write(project_vals, context=ctx)
+
+        if context is None:
+            context = dict()
+        context['from_event'] = True

```

Comment:
This is no longer a copy of context. Use carefully or the context will contain a lot of unneeded things (when going back to caller).